tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -496,40 +497,54 @@ class FirewallRulesFilter(Filter):
                             - 10.20.20.0/24
     """
 
-    schema = type_schema(
-        'firewall-rules',
-        **{
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['firewall-rules']},
             'include': {'type': 'array', 'items': {'type': 'string'}},
+            'any': {'type': 'array', 'items': {'type': 'string'}},
+            'only': {'type': 'array', 'items': {'type': 'string'}},
             'equal': {'type': 'array', 'items': {'type': 'string'}}
-        })
+        },
+        'oneOf': [
+            {"required": ["type", "include"]},
+            {"required": ["type", "any"]},
+            {"required": ["type", "only"]},
+            {"required": ["type", "equal"]}
+        ]
+    }
+
     schema_alias = True
 
     def __init__(self, data, manager=None):
         super(FirewallRulesFilter, self).__init__(data, manager)
         self.policy_include = None
         self.policy_equal = None
+        self.policy_any = None
```

Comment:
Please update docs

Reply:
I was having trouble finding any but now I see that a few of the individual resources had a few lines where they listed the filter.  I'll move something into the core filter so it auto-gens correctly.

Reply:
Ok, I think I've got that right.